lib/backend/syncersv1/updateprocessors/felixnodeprocessor.go

```diff
@@ -93,6 +93,19 @@ func (c *FelixNodeUpdateProcessor) Process(kvp *model.KVPair) ([]*model.KVPair,
 				}
 			}
 		}
+		// Look for internal node address, if BGP is not running
```

Comment:
@fasaxc @sridhartigera I want to check my understanding here.

IIRC, newer version of Felix and Typha ignore the HostIPKey and instead use the full v3 Node object.

e.g. 
- https://github.com/projectcalico/felix/blob/c46999eec127ecec7dec03f5f23201e1d6eff987/calc/vxlan_resolver.go#L82-L83
- https://github.com/projectcalico/felix/blob/bbd5c2b04230b1f23006ad44fab1ed6ef79e1270/calc/l3_route_resolver.go#L160-L166

So I think in practice we could do without these lines? The only reason these would be needed is if we're running a new version of Typha against a vey old version of Felix, since the new Typha would need to send data model updates in a way that the old felix understands.

Reply:
I think the main thing I want to make sure is that any new code we're writing on the other side of this in Felix is looking at the v3 object and not at the HostIPKey, since that would be a step in the wrong direction.

Reply:
We still use the HostIPKey 
https://github.com/projectcalico/felix/blob/master/calc/dataplane_passthru.go#L43. and the BPF end point manager receives HostMetaDataUpdate message from eventSequencer.

Reply:
sad.  I guess we're stuck with this for now.

It would be nice to clean up that code to use the full v3 resource. At that point we could stop mucking around with HostIPKeys in this code so we don't need to send the same information in multiple formats.

Reply:
So I think that means the CalicoNodeIP type is essentially unused until we switch the Felix code over to use the v3 resource.

Once Felix is switched, felix can determine which IP to use based on the list of addresses rather than the HostIPKey.

Reply:
Right, CalicoNodeIP is unused for now.